packages/contracts-bedrock/src/L2/L1Block.sol

```diff
@@ -73,4 +85,34 @@ contract L1Block is ISemver {
         l1FeeOverhead = _l1FeeOverhead;
         l1FeeScalar = _l1FeeScalar;
     }
+
+    /// @notice Updates the L1 block values for a post-blob activated chain.
+    /// Params are passed in as part of msg.data in order to compress the calldata.
+    /// Params should be passed in in the following order:
+    ///   1. _basefeeScalar      L1 basefee scalar
+    ///   2. _blobBasefeeScalar  L1 blob basefee scalar
+    ///   3. _sequenceNumber     Number of L2 blocks since epoch start.
+    ///   4. _number             L1 blocknumber.
+    ///   5. _timestamp          L1 timestamp.
+    ///   6. _basefee            L1 basefee.
+    ///   7. _blobBasefee        L1 blobBasefee.
+    ///   8. _hash               L1 blockhash.
+    ///   9. _batcherHash        Versioned hash to authenticate batcher by.
+    function setL1BlockValuesEcotone() external {
+        require(msg.sender == DEPOSITOR_ACCOUNT, "L1Block: only the depositor account can set L1 block values");
+        uint256 data;
+        assembly {
+            data := calldataload(4)
+            sstore(basefeeScalar.slot, shr(192, data)) // basefeeScalar & blobBaseFeeScalar (uint32,uint32)
+			// sequenceNumber is a 64 bit int that has a 256-bit slot to itself, so we shift-left
+			// the data 64 bits to clear out the scalar bits before shifting right, ensuring the
+			// topmost 192 bits in the slot will always be 0.
+            sstore(sequenceNumber.slot, shr(192, shl(64, data))) // uint64
+            sstore(number.slot, shr(128, calldataload(20))) // number & timestamp (uint64,uint64)
+            sstore(basefee.slot, calldataload(36)) // uint256
+            sstore(blobBasefee.slot, calldataload(68)) // uint256
+            sstore(hash.slot, calldataload(100)) // bytes32
+            sstore(batcherHash.slot, calldataload(132)) // bytes32
+        }
+    }
```

Comment:
Would prefer a few changes to this function:
1. `require` is way too expensive for a function that's called every block; would prefer if we move this check into inline assembly and XOR the addresses.
2. `data` does not need to be allocated on the stack in solidity, can do in asm.
3. I prefer a shift -> mask approach when isolating a value in the middle of the word.

Also, please add test coverage for the `msg.sender == DEPOSITOR_ACCOUNT` branch.

```suggestion
    function setL1BlockValuesEcotone() external {
        assembly {
            // Revert if the caller is not the depositor account.
            if xor(caller(), DEPOSITOR_ACCOUNT) {
                mstore(0x00, 0x3cc50b45)
                revert(0x1C, 0x04)
            }

            let data := calldataload(4)
            sstore(basefeeScalar.slot, shr(192, data)) // basefeeScalar & blobBaseFeeScalar (uint32,uint32)
			// sequenceNumber is a 64 bit int that exists in the [128, 192) bits of the first word of calldata. We
            // shift it right by 128 and then mask off the low-order 64 bits to ensure that the high-order 192 bits
            // of the slot are cleaned.
            sstore(sequenceNumber.slot, and(shr(128, data), 0xFFFFFFFFFFFFFFFF)) // uint64
            sstore(number.slot, shr(128, calldataload(20))) // number & timestamp (uint64,uint64)
            sstore(basefee.slot, calldataload(36)) // uint256
            sstore(blobBasefee.slot, calldataload(68)) // uint256
            sstore(hash.slot, calldataload(100)) // bytes32
            sstore(batcherHash.slot, calldataload(132)) // bytes32
        }
    }
```

Reply:
done!